presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestCachingHiveMetastore.java

```diff
@@ -208,19 +242,50 @@ public void testCachingWithPartitionVersioning()
         assertEquals(partitionCachingEnabledmetastore.getPartitionsByNames(TEST_DATABASE, TEST_TABLE, ImmutableList.of(TEST_PARTITION1, TEST_PARTITION2)).size(), 2);
         assertEquals(mockClient.getAccessCount(), 3);
 
-        // This call should invalidate the partition cache because when partition versioning is enabled
-        // partitions with older or empty version will be deleted from cache. And the hivePartitionVersionFetcher used in creating
-        // partitionCachingEnabledmetastore will set the version as Optional.empty() for all the partitions fetched.
+        // This call should NOT invalidate the partition cache because partition version is same as before
         assertEquals(partitionCachingEnabledmetastore.getPartitionNamesByFilter(TEST_DATABASE, TEST_TABLE, ImmutableMap.of()), expectedPartitions);
         assertEquals(mockClient.getAccessCount(), 4);
 
         assertEquals(partitionCachingEnabledmetastore.getPartitionsByNames(TEST_DATABASE, TEST_TABLE, ImmutableList.of(TEST_PARTITION1, TEST_PARTITION2)).size(), 2);
-        // Assert that the previous cache entry is invalidated and that we did not hit the cache
-        assertEquals(mockClient.getAccessCount(), 5);
+        // Assert that its a cache hit
+        assertEquals(mockClient.getAccessCount(), 4);
```

Comment:
Can we extract the following two tests for increasing and decreasing to a common function and use increasing/decreasing as the argument?